CPP_STYLE_GUIDE.md

```diff
@@ -257,6 +258,24 @@ class ExampleClass {
 };
 ```
 
+### When to use AliasedBuffers
+
+When working with typed arrays that involves direct data modification
+from C++, and when such modifications can have side effects outside
+of the callsite, use an AliasedBuffer when possible, because that
```

Comment:
Sorry, but I’m still somewhat confused… What does 

> when such modifications can have side effects outside of the callsite

mean?

Reply:
I mean to express this:
If the C++ layer modification of data is unanticipated / unforeseen / invisible at the callsite.

Example:
```js
buffer.fill('x')
```
here, the backend buffer is filled with data at C++ level, and this is evident by looking at the JS callsite. Whereas:

```js
obj.registerModule('foo)';
```
running into C++ land, and modifying artifacts linked to `obj` is not evident at the callsite.

My way of telling is to recommend AliasedBuffer usage in second type of cases. Does it make any meaning to you?

Reply:
@gireeshpunathil Yes, but … the thing is, afaik (and you or @nodejs/chakracore can correct me if I’m wrong), there’s no real difference between these cases.

The VM debugger wouldn’t be notified of modifications even in the `buffer.fill('x')` case; and because we don’t use AliasedBuffer (and we can’t really, unless we port memset/memcpy to it), they still have to do explicit notifications after our modifications:

https://github.com/nodejs/node-chakracore/blob/38520056313bc747e3ca6dcc7598c2e51394c0ac/src/node_buffer.cc#L575-L583

(They haven’t fully covered all the `Buffer` methods, to be fair.)

Reply:
@addaleax - ok; I removed that phrase to avoid confusion.

[On a side note]: In a sufficiently large application, what the VM (or a monitoring service that leverages the VM function) observed data at this level will be useful for? In other words, for a program or a user, a `buffer.fill` is an obvious callsite that mutates the buffer data at some lower levels, so may not be of real interest (or of any use), but would be interested to get notified on an out-of-bounds access because there is no normal way of knowing it otherwise?

Reply:
@gireeshpunathil So, the ChakraCore application for this whole notification thingy is time-travel debugging… I haven’t ever tried that myself, but I believe the purpose here is to be able to compare the values of JS objects at different points in time, and that’s what the notifications are for? Without this, I think the debugger could show the same contents of a buffer after a `.fill()` call and when going back in time to before that call?